app-backend/api/src/main/scala/user/Auth0User.scala

```diff
@@ -323,32 +349,142 @@ object Auth0Service extends Config with LazyLogging {
+    } else if (numTries > 15) {
+      Future {
+        val e =
+          BulkJobProcessTimeout("Exhausted tries when bulk creating users")
+        logger.error(e.error)
+        Left(e)
+      }
+    } else {
+      // Wait one second before requesting to not spam Auth0
+      Thread.sleep(1000)
+      basicRequest
+        .header("Authorization", s"Bearer ${bearerToken.access_token}")
+        .get(uri"https://$auth0Domain/api/v2/jobs/$jobId")
+        .response(asJson[Auth0JobStatus])
+        .send()
+        .flatMap { response =>
+          response.body match {
+            case Left(_) =>
+              Future {
+                val e = BulkJobSubmitError(
+                  s"Error Processing: ${response.code} - ${response.statusText}"
+                )
+                logger.error(e.error)
+                Left(e)
+              }
+            case Right(status) =>
+              status.status match {
+                case "completed" =>
+                  waitForJobToComplete(jobId, true, numTries, bearerToken)
+                case _ =>
+                  waitForJobToComplete(
+                    jobId,
+                    jobComplete,
+                    numTries + 1,
+                    bearerToken
+                  )
+              }
+          }
+        }
+    }
+  }
 
-    val managementBearerHeaders = getBearerHeaders(bearerToken)
+  def getBulkCreatedUsers(
+      bulkCreateId: String
+  ): Future[Either[BulkCreateError, List[Auth0User]]] = {
+    val q = s"""app_metadata.bulkCreateId:"$bulkCreateId""""
+    for {
+      managementToken <- authBearerTokenCache.get(1)
+      requestUri = uri"https://$auth0Domain/api/v2/users?q=$q&per_page=100"
+      authHeader = s"Bearer ${managementToken.access_token}"
+      response <- basicRequest
+        .header("Authorization", authHeader)
+        .get(requestUri)
+        .response(asJson[List[Auth0User]])
+        .send()
+        .map { r =>
+          r.body match {
+            case Left(_) => {
+              val e = BulkJobRequestUsersError(
+                s"Status Code: ${r.code}, Status Text: ${r.statusText} for $requestUri"
+              )
+              logger.error(e.error)
+              Left(e)
+            }
+            case Right(r) => Right(r)
+          }
+        }
+    } yield response
 
-    Http()
-      .singleRequest(
-        HttpRequest(
-          method = POST,
-          uri = s"$userUri",
-          headers = managementBearerHeaders,
-          entity = HttpEntity(
-            ContentTypes.`application/json`,
-            post.noSpaces
-          )
+  }
+
+  def bulkCreateUsers(
+      userIds: List[String]
+  ): Future[Either[BulkCreateError, List[Auth0User]]] = {
+    val bulkCreateId = UUID.randomUUID()
+    val usersToCreate = userIds
+      .map(
+        uid =>
+          Auth0UserBulkCreate(
+            uid,
+            s"$uid@${auth0AnonymizedConnectionName}.com",
+            true,
+            uid.bcrypt(10),
```

Comment:
I still don't really understand how passwords work, but according to Auth0 docs we need to supply the hashed password we want to use.